internal/gardenclient/client.go

```diff
@@ -257,7 +262,34 @@ func (g *clientImpl) GetConfigMap(ctx context.Context, namespace, name string) (
 	return cm, nil
 }
 
+func (g *clientImpl) GetManagedSeed(ctx context.Context, name string) (*seedmanagementv1alpha1.ManagedSeed, error) {
+	managedSeed := &seedmanagementv1alpha1.ManagedSeed{}
+	key := types.NamespacedName{Name: name, Namespace: "garden"} // Currently, managed seeds are restricted to the garden namespace
+
+	if err := g.c.Get(ctx, key, managedSeed); err != nil {
+		return nil, fmt.Errorf("failed to get managed seed %v: %w", key, err)
+	}
+
+	return managedSeed, nil
+}
+
 func (g *clientImpl) GetSeedClientConfig(ctx context.Context, name string) (clientcmd.ClientConfig, error) {
+	managedSeed, err := g.GetManagedSeed(ctx, name)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return nil, err
+	}
+
+	if managedSeed != nil {
+		clientConfig, err := g.GetShootClientConfig(ctx, managedSeed.Namespace, managedSeed.Name)
+		if err != nil && !apierrors.IsNotFound(err) {
+			return nil, err
+		}
+
+		if clientConfig != nil {
+			return clientConfig, nil
+		}
+	}
```

Comment:
```suggestion
	if shootName, err := g.GetShootNameofSeed(ctx, name); err != nil {
		return nil, err
	} else if shootName != nil {
	        // I would prefer the following
	        // return g.GetShootClientConfig(ctx, "garden", *shootName)
	        clientConfig, err := g.GetShootClientConfig(ctx, "garden", *shootName)
	        if err == nil {
                       return clientConfig, nil
                } else if !apierrors.IsNotFound(err) {
                       return nil, err
                }
	}
```

Reply:
I incorporated  the changes. We still need to discuss the fallback to seed kubeconfig in case of an error.